packages/react/src/components/login/apiTokensClient/apiTokenClient.test.ts

```diff
@@ -0,0 +1,555 @@
+    });
+    it('updates after a new fetch', async () => {
+      const secondJSON = { token: 'a token' };
+      const apiTokenClient = await initTestsWithDefaultPropsAndWaitForInit({
+        additionalResponses: [{ returnedStatus: HttpStatusCode.OK, body: JSON.stringify(secondJSON) }],
+      });
+      expect(apiTokenClient.getTokens()).toMatchObject(apiTokens);
+      listenToPromise(apiTokenClient.fetch(currentUser as User));
+      await waitForFetchMockResultFulfillment(1);
+      expect(currentApiTokenClient.getTokens()).toMatchObject(secondJSON);
+    });
+  });
+  describe(`.clear()`, () => {
+    it('Removes current tokens, also from storage. User reference is also removed. API_TOKENS_REMOVED event is emitted.', async () => {
+      const removeSpy = jest.spyOn(Storage.prototype, 'removeItem');
+      const apiTokenClient = await initTestsWithDefaultPropsAndWaitForInit();
+      expect(removeSpy).toHaveBeenCalledTimes(4);
+      expect(apiTokenClient.getTokens()).not.toBeNull();
+      apiTokenClient.clear();
+      expect(apiTokenClient.getTokens()).toBeNull();
+      expect(getUserReferenceFromStorage()).toBeNull();
+      expect(removeSpy).toHaveBeenCalledTimes(6);
+      expect(getAllMockCallArgs(removeSpy)[0][0]).toBe(API_TOKEN_SESSION_STORAGE_KEY);
+      expect(getAllMockCallArgs(removeSpy)[1][0]).toBe(API_TOKEN_SESSION_USER_REFERENCE_KEY);
+      expect(getEmittedErrors()).toHaveLength(0);
+      await waitFor(() => {
+        const eventPayloads = getEmittedEventPayloads();
+        // renewal events are #0 and #1
+        expect(eventPayloads).toHaveLength(3);
+        expect(eventPayloads[2].type).toBe(apiTokensClientEvents.API_TOKENS_REMOVED);
+        expect(eventPayloads[2].data).toBeNull();
+      });
+    });
+  });
+  describe(`When init signal is received `, () => {
+    it('and user matches api token reference, tokens are restored from storage. API_TOKENS_UPDATED event is emitted', async () => {
+      const tokens = { token: 'a token' };
+      const userAccessToken = 'an access token';
+      setApiTokensToStorage(tokens);
+      setUserReferenceToStorage(userAccessToken);
+      const apiTokenClient = initTests({
+        apiTokenClientProps: { maxRetries: 0 },
+        userProps: { signInResponseProps: { access_token: userAccessToken } },
+      });
+      expect(apiTokenClient.getTokens()).toMatchObject(tokens);
+      expect(getRequestCount()).toBe(0);
+      await waitFor(() => {
+        const eventPayloads = getEmittedEventPayloads();
+        expect(eventPayloads).toHaveLength(1);
+        expect(eventPayloads[0].type).toBe(apiTokensClientEvents.API_TOKENS_UPDATED);
+        expect(eventPayloads[0].data).toMatchObject(tokens);
+      });
+      expect(apiTokenClient.isRenewing()).toBeFalsy();
+    });
+    it('and user does not match the api token reference, tokens are cleared and fetched. User reference is stored. API_TOKENS_UPDATED and API_TOKENS_RENEWAL_STARTED events are emitted.', async () => {
```

Comment:
Could the test name be shorter?

Reply:
Shortened. BTW, why test names cannot be long? Best place to tell clearly what is tested and expected.

Reply:
I feel like long test names clutter the test feed when running them. This could be a me problem, but when a test name has all these things that it's checking, by the end of the paragraph I've forgotten everything except the last two sentences. I find it much more understandable when tests are like functions: doing one thing and one thing only. If that's not possible and the test is checking many things, like here, personally I really do not mind if the test name is written in a general fashion and there are comments in the test. Because if the test fails, I see the name and get a general understanding of what it's doing, and then I'll open the test file and see what's going on there and read the comments if there are any.

This might be just personal views clashing. Since you argumented well why long test names are good before, I can respect it and we can keep them 😄

Reply:
Ok. I didn't do any changes for this.